broadinstitute/single_cell_portal_core
# Remove modal and renew user access token (SCP-5164)

This updates the handling for a user's access token expiring. Previously we popped open a modal telling users to re-sign-in if they were logged in but the `window.SCP.userAccessToken` was blank, this modal was not helpful because the user didn't actually need to re-sign-in they would just need to refresh the page. This change follows the logic utilized for the `readOnlyToken` and renews the `userAccessToken` before it expires and removes the modal entirely so the user should not be bothered or notice the token getting refreshed as appropriate.

To test:
You can check out the new API endpoint in swagger - https://localhost:3000/single_cell/api/v1#/Site/site_renew_user_access_token 

If you don't authorize you won't get a token, but if you authorize before trying it you will see the returned token and expiration info.

You can do a semi-proper test locally by going to line 268 in `scp-api.jsx` and change the renewalTime to something much quicker like 60000 (1 minute). 
Then:
- boot up your localhost
- watch the rails log for something like after the allotted time you set for the timeout above:

> Started GET "/single_cell/api/v1/site/renew_user_access_token" for 127.0.0.1 at 2023-08-02 14:20:58 -0400
Processing by Api::V1::SiteController#renew_user_access_token as JSON
Renewing user access token via SCP API for #<User:0x00000002853f5e80>
Completed 200 OK in 1550ms (Views: 0.7ms | MongoDB: 0.8ms | Allocations: 61357)
